### DISCUSSION TITLE
Combobox.Button wrapper Combobox.Input auto close when I typing space

### DISCUSSION THREAD

**Discussion by thiennv-ns** [tailwindlabs/headlessui#1209] [Help]
<Combobox.Button className={isMainLayout ? 'mr-2' : 'mr-4'}>
              <Combobox.Input
                value={query}
                placeholder={placeholder}
              />
   </Combobox.Button>

   How I can remove the keypress space for Combobox.Button ?

**Comment by jdreic** [comment]
I ran into this as well - if you render the `Combobox.Button` and `Combobox.Input` as siblings, rather than having the `Combobox.Input` as a child of the `Combobox.Button`, then you won't have this issue.

**Comment by RobinMalfait** [comment]
Hey! Don't render an input inside a button that doesn't make much sense.

That's like doing this in html:
```html
<button>
  <input />
</button>
```

The button can be used as a sibling to render a little dropdown icon or similar.

What's the reason you want to do this?

**Comment by roydigerhund** [comment]
I did the same thing in order to have the options opened as soon as the input is clicked. A workaround would be to render the button as `div`:  

`<Combobox.Button as="div">`

Other approaches can be seen here https://github.com/tailwindlabs/headlessui/discussions/1236

**Reply by sipec** [reply]
As of https://github.com/tailwindlabs/headlessui/pull/2686 you don't need to do this hack anymore. just add the prop `immediate`
